mattermost-plugin/server/plugin.go

```diff
@@ -103,7 +113,21 @@ func (p *Plugin) OnActivate() error {
 		EnableLocalMode:         false,
 		LocalModeSocketLocation: "",
 		AuthMode:                "mattermost",
-	}, "", logger)
+	}
+	var db store.Store
+	db, err = sqlstore.New(cfg.DBType, cfg.DBConfigString, cfg.DBTablePrefix, logger, sqlDB)
```

Comment:
Maybe we can pass to the `NewStore` constrator the `sqlDB` and that way this gets simplified too.

Reply:
Hmm .. yeah I thought about that. It would cause "sql.Open + db.Ping" code to be repeated in 4 places. I think a reasonable cleanup would be to always pass `sql.DB` to sqlstore.New and move the db creation into `NewStore`. That way the `if db == nil` in `sqlstore.New` logic gets cleared up.

So the only difference remains in the plugin path and server path. That keeps things separate enough.